### PR TITLE
Update Thunderbird Privacy Policy

### PR DESCRIPTION
**Pull request by Alecaddd**
- Specify privacy policy applicable to the most recent Thunderbird distributions.
- Highlight that Mozilla does not retain any sensitive data during OAuth process.
- Add a paragraph about email and calendar data storage.
- Add "Contact Us" section with postal information and link to Data Subject Access Requests.

### CODE REVIEW

**Review by peiying2 — CHANGES_REQUESTED** [review]
Minor change request.

File: en/thunderbird_privacy_policy.md
Comment (peiying2) [reply en/thunderbird_privacy_policy.md:104]: @mafeld please have a look.

File: en/thunderbird_privacy_policy.md
Comment (flodolo) [inline en/thunderbird_privacy_policy.md:118]: At risk of pointing out the obvious, this line break won't have much effect in markdown, it will just become an empty space after `removed.`.

File: en/thunderbird_privacy_policy.md
Comment (Alecaddd) [reply en/thunderbird_privacy_policy.md:104]: As @flodolo wrote, Mailfence doesn't have a URL redirect for that link, sorry.

I would like to kindly ask to not block this PR with change requests to sections that haven't been touched and focus only on the modified parts.
We need these changes to unblock Google OAuth API review for K-9 and the changes have already been reviewed by the MoCo lawyer.

File: en/thunderbird_privacy_policy.md
Comment (Alecaddd) [reply en/thunderbird_privacy_policy.md:118]: Good catch, I didn't notice that

Comment (flodolo) [reply en/thunderbird_privacy_policy.md:118]: Do you want me to fix it or publish it as is? Just line break or new paragraph?

Comment (Alecaddd) [reply en/thunderbird_privacy_policy.md:118]: If you could fix it and publish it it would be great.
Thank you so much!

Comment (Alecaddd) [reply en/thunderbird_privacy_policy.md:118]: A new paragraph is good, thanks

Comment (flodolo) [reply en/thunderbird_privacy_policy.md:118]: Looks like I don't permissions to push to your branch (it's also on main, so you might want to remove the fork and refork it). I'll fix after the merge.

**Pull request merged**
